automated install/basic-install.sh

```diff
@@ -2814,7 +2813,6 @@ main() {
 
     # Remaining log files
     if [ -f /var/log/pihole.log ] && [ ! -L /var/log/pihole.log ]; then
-        mkdir -p /var/log/pihole/
         mv /var/log/pihole*.* /var/log/pihole/ 2>/dev/null
     fi
```

Comment:
My local repo did not have the special handling lines when I did the hasty hotfix. This PR is the correct place.